lib/samples/show_grid/show_grid_sample.dart

```diff
@@ -0,0 +1,368 @@
+    if (_mapViewController.grid != null) {
+      final grid = _mapViewController.grid!;
+      grid.labelPosition = labelPosition;
+    }
+  }
+
+  /// Show the grid options dialog.
+  void _showGridOptions() async {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Grid Options'),
+          content: GridOptions(
+            onGridTypeChanged: _onGridTypeChanged,
+            onGridColorTypeChanged: _onGridColorTypeChanged,
+            onLabelColorTypeChanged: _onLabelColorTypeChanged,
+            onLabelPositionTypeChanged: _onLabelPositionTypeChanged,
+            onLabelVisibilityChanged: _onLabelVisibilityChanged,
+          ),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('Close'),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+///
+/// a widget that displays grid options.
+///
+class GridOptions extends StatefulWidget {
+  final Function(Grid) onGridTypeChanged;
+  final Function(Color) onGridColorTypeChanged;
+  final Function(Color) onLabelColorTypeChanged;
+  final Function(GridLabelPosition) onLabelPositionTypeChanged;
+  final Function(bool) onLabelVisibilityChanged;
+
+  GridOptions({
+    Key? key,
+    required this.onGridTypeChanged,
+    required this.onGridColorTypeChanged,
+    required this.onLabelColorTypeChanged,
+    required this.onLabelPositionTypeChanged,
+    required this.onLabelVisibilityChanged,
+  }) : super(key: key);
+
+  @override
+  State<GridOptions> createState() => _GridOptionsState();
+}
+
+class _GridOptionsState extends State<GridOptions> {
+  final gridTypes = ['Latitude & Longitude', 'MGRS', 'UTM', 'USNG'];
+  final colorTypes = ['Red', 'Blue', 'Green', 'Yellow'];
+  final labelPositionTypes = [
+    'AllSides',
+    'BottomLeft',
+    'BottomRight',
+    'Center',
+    'Geographic',
+    'TopLeft',
+    'TopRight'
+  ];
+
+  // stateful variables
+  String? gridType;
+  String? gridColorType;
+  String? labelColorType;
+  String? labelPositionType;
+  bool labelVisible = true;
+
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollView(
+      child: ListBody(
+        children: <Widget>[
+          _buildGridTypeDropdown(),
+          _buildGridColorTypeDropdown(),
+          _buildLabelColorTypeDropdown(),
+          _buildLabelPositionTypeDropdown(),
+          _buildLabelVisibilityCheckbox(),
+        ],
+      ),
+    );
+  }
+
+  // Create a DropdownButtonFormField widget.
+  static DropdownButtonFormField _createDropdownButtonFormField({
+    required String value,
+    required String labelText,
+    required List<String> items,
+    required Function onChanged,
+  }) {
+    return DropdownButtonFormField<String>(
+      value: value,
+      onChanged: (String? newValue) {
```

Comment:
```suggestion
      onChanged: (newValue) {
```
I think this can be inferred and we prefer to infer where it's possible (and doesn't just infer `dynamic`)